ql/src/test/org/apache/hadoop/hive/ql/io/parquet/read/TestDataWritableReadSupport.java

```diff
@@ -69,4 +69,23 @@ public void testGetProjectedSchema3() throws Exception {
       .getProjectedSchema(originalMsg, Arrays.asList("structCol", "c"), Arrays.asList(0, 1),
         Arrays.asList("structCol.b", "c")).toString());
   }
+
+  @Test
+  public void testGetProjectedSchema4() throws Exception {
+    MessageType originalMsg = MessageTypeParser.parseMessageType(
+      "message hive_schema {\n"
+        + "  optional group structCol {\n"
+        + "    optional int32 a;\n"
+        + "    optional group subStructCol {\n"
+        + "      optional int64 b;\n"
+        + "      optional boolean c;\n"
+        + "    }\n"
+        + "  }\n"
+        + "  optional boolean d;\n"
+        + "}\n");
+
+    testConversion("structCol", "struct<subStructCol:struct<b:bigint>>", DataWritableReadSupport
+      .getProjectedSchema(originalMsg, Arrays.asList("structCol"), Arrays.asList(0),
+        Arrays.asList("structCol.subStructCol.b")).toString());
```

Comment:
Hmm.. this is not what I mean. I was thinking the case: `Arrays.asList("structCol.subStructCol", "structCol.subStructCol.b")`. Does this make sense?